### PR TITLE
[query] support describe statement #1028

### LINKED ISSUE #1028: [query] DESCRIBE Statement
**Summary**

`DESC|DESCRIBE tbl_name`

Reference:
https://dev.mysql.com/doc/refman/8.0/en/describe.html
https://clickhouse.tech/docs/en/sql-reference/statements/describe-table/

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

support `DESC|DESCRIBE tbl_name`.

The result based on https://github.com/datafuselabs/datafuse/blob/master/common/datavalues/src/data_field.rs#L11

```
mysql> create table t(a int , b float);
Query OK, 0 rows affected (0.00 sec)

mysql> describe t;
+-------+---------+------+
| Field | Type    | Null |
+-------+---------+------+
| a     | Int32   | NO   |
| b     | Float32 | NO   |
+-------+---------+------+
2 rows in set (0.00 sec)
```

## Changelog

- New Feature

## Related Issues

Fixes #1028 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by zhyass** [comment]
TODO: add unit test

**Comment by BohuTANG** [comment]
> TODO: add unit test
> 
> I will complete the unit test case in spare time.

Cool, and the stateless test :100:

**Comment by zhyass** [comment]
> > TODO: add unit test
> > I will complete the unit test case in spare time.
> 
> Cool, and the stateless test 💯

Done

File: fusequery/query/src/interpreters/interpreter_describe_table.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+#[async_trait::async_trait]
+impl Interpreter for DescribeTableInterpreter {
+    fn name(&self) -> &str {
+        "DescribeTableInterpreter"
+    }
+
+    async fn execute(&self) -> Result<SendableDataBlockStream> {
+        let datasource = self.ctx.get_datasource();
```
Comment (zhang2014) [inline fusequery/query/src/interpreters/interpreter_describe_table.rs:37]: maybe `self.ctx.get_table(db, table)` is better ?

File: fusequery/query/src/interpreters/interpreter_describe_table.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        let table = database.get_table(self.plan.table.as_str())?;
+        let schema = table.schema()?;
+
+        let mut names: Vec<String> = vec![];
+        let mut types: Vec<String> = vec![];
+        let mut nulls: Vec<String> = vec![];
+        for field in schema.fields().iter() {
+            names.push(field.name().to_string());
+            types.push(format!("{:?}", field.data_type()));
+            nulls.push(if field.is_nullable() {
```
Comment (zhang2014) [inline fusequery/query/src/interpreters/interpreter_describe_table.rs:48]: maybe use `BoolSeries` is better? (true or false)

File: fusequery/query/src/sql/plan_parser.rs [hunk 3]
```diff
@@ -270,6 +273,32 @@ impl PlanParser {
...
+        let mut table = describe.name.0[0].value.clone();
+        if describe.name.0.len() > 1 {
+            db = table;
+            table = describe.name.0[1].value.clone();
+        }
+
+        let schema = DataSchemaRefExt::create(vec![
+            DataField::new("Field", DataType::Utf8, false),
+            DataField::new("Type", DataType::Utf8, false),
+            DataField::new("Null", DataType::Utf8, false),
```
Comment (zhang2014) [inline fusequery/query/src/sql/plan_parser.rs:292]: maybe `DataType::Uft8 -> DataType::Bool` is better

File: website/datafuse/docs/sqlstatement/describe-commands/describe-table.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
```
Comment (zhang2014) [inline website/datafuse/docs/sqlstatement/describe-commands/describe-table.md:1]: Great job!

File: fusequery/query/src/interpreters/interpreter_describe_table.rs (hunk 2 above)
Comment (zhyass) [reply fusequery/query/src/interpreters/interpreter_describe_table.rs:48]: Here reference the Mysql syntax
https://dev.mysql.com/doc/refman/8.0/en/show-columns.html

Comment (zhang2014) [reply fusequery/query/src/interpreters/interpreter_describe_table.rs:48]: Understand. mysql style is good.

File: fusequery/query/src/interpreters/interpreter_describe_table.rs (hunk 1 above)
Comment (zhyass) [reply fusequery/query/src/interpreters/interpreter_describe_table.rs]: Good suggestion, done.

**Review by zhang2014 — APPROVED** [review]
LGTM

**Pull request merged**